fofix/core/GameEngine.py

```diff
@@ -292,29 +289,94 @@ def __init__(self, config = None):
 
         self.setlistMsg = None
 
+        # Init theme system
+        themename = self.config.get("coffee", "themename")
+        themepath  = os.path.join(Version.dataPath(), "themes", themename)
+        self._initTheme(themename, themepath)
 
         # Load game modifications
         Mod.init(self)
-        self.task.addTask(self.input, synced = False)
 
+        self.task.addTask(self.input, synced = False)
         self.task.addTask(self.view, synced = False)
-
         self.task.addTask(self.resource, synced = False)
 
         self.data = Data(self.resource, self.svg)
 
+        self.input.addKeyListener(FullScreenSwitcher(self), priority = True)
+        self.input.addSystemEventListener(SystemEventHandler(self))
+
+        self.debugLayer         = None
+        self.startupLayer       = None
+        self.loadingScreenShown = False
+        self.graphicMenuShown   = False
+
+        log.debug("Ready.")
+
+
+    # evilynux - This stops the crowd cheers if they're still playing (issue 317).
+    def quit(self):
+        # evilynux - self.audio.close() crashes when we attempt to restart
+        if not self.restartRequested:
+            self.audio.close()
+        Player.savePlayers()
+        for taskData in list(self.task.tasks):
+            self.task.removeTask(taskData['task'])
+        self.running = False
+
+
+    def _initTheme(self, themename, themepath):
+        '''
+        Select the source of graphics for the game.
+        Note that currently this can only be called GameEngine on startup.
+
+        :param themename: what to call this theme
+        :type themename: str
+        :param themepath: absolute path to theme folder
+        :type themepath: str
+        '''
+
+        log.notice( 'Setting theme %s from "%s"' % (themename,themepath) )
+
+        self.theme = None
+        try:
+            # Look for "CustomTheme.py" inside theme dir
+            fp, pathname, description = imp.find_module("CustomTheme",[themepath])
+            try:
+                # Found it! Load it.
+                theme = imp.load_module("CustomTheme", fp, pathname, description)
```

Comment:
Just a note while imp.load_module still exists in python 3+ but it is now depreciated.
https://docs.python.org/3.6/library/imp.html#imp.load_module

Reply:
Doesn't really change anything here, just something to be aware of if we are moving to python 3 at some point.

Reply:
Are you volunteering? 😀 BTW, I didn't write that code, I just rearranged the try/except stuff.

Reply:
Yes, this method is deprecated, and this is why I suggested to use only one `try / except` block (as suggested in the doc) ;). But we should keep this method for now.
(I'm already working on porting the code to python3 and this is not a problem ;).)